project/schema.go

```diff
@@ -0,0 +1,286 @@
+package project
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/docker/docker/pkg/nat"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+var fieldsSchemaLoader gojsonschema.JSONLoader
+var serviceSchemaLoader gojsonschema.JSONLoader
```

Comment:
@joshwget I think you can go one of two ways with this.  Either call `setupSchemaLoaders()` from `init()` such that it's done once, or call `setupSchemaLoaders()` every time and not cache at the package level.  Currently this code will suffer from (albeit unlikely) race conditions.  I would prefer you call it from `init()`.  From a server context I'd rather initialize this structure once.

Can you confirm that gojsonschema.JSONLoader can be used concurrently across goroutines?